pyannote/metrics/diarization.py

```diff
@@ -792,3 +794,207 @@ def compute_components(self,
+WDER_SPEAKER_SWITCH_TOTAL = 'Word Diarization Error Rate Speaker Switch Total Words'
+
+
+class WordDiarizationErrorRate(IdentificationErrorRate):
+    """Word diarization error rate
+
+    First, the optimal mapping between reference and hypothesis labels
+    is obtained using the Hungarian algorithm. For each word, the diarization system’s assigned
+    speaker label is compared to the reference speaker label. If the assigned speaker is incorrect,
+    it’s considered an error.
+
+    WDER is calculated as:
+        WDER = (SIS + CIS) / (S + C)
+        where,
+            SIS = substitutions with incorrect speaker label
+            CIS = correct ASR words with incorrect speaker label
+            S   = number of ASR substitutions
+            C   = number of correct ASR words
+
+    Reference Transcript and Speaker Labels
+
+    | Word  | Reference Speaker |
+    |-------|-------------------|
+    | Hello | Speaker 1         |
+    | How   | Speaker 1         |
+    | are   | Speaker 1         |
+    | you   | Speaker 1         |
+    | today | Speaker 2         |
+
+    Diarization System's Output
+
+    | Word  | Diarization Output Speaker |
+    |-------|----------------------------|
+    | Hello | Speaker 1                  |
+    | How   | Speaker 2                  |
+    | are   | Speaker 1                  |
+    | you   | Speaker 1                  |
+    | today | Speaker 1                  |
+
+    In this example, the word “How” was incorrectly assigned to Speaker 2 instead of Speaker 1,
+    and the word “today” was incorrectly assigned to Speaker 1 instead of Speaker 2.
+    These are the two errors that contribute to the Word Diarization Error Rate (WDER) calculation.
+    WDER = 2 / 5 = 0.4. This means that 40% of the words were assigned to the wrong speaker.
+
+    Usage
+    ----------
+
+    * Word diarization error rate between `reference` and `hypothesis` annotations
+
+        >>> metric = WordDiarizationErrorRate()
+        >>> reference = Annotation(...)                 # doctest: +SKIP
+        >>> reference_words = Annotation(...)           # doctest: +SKIP
+        >>> hypothesis = Annotation(...)                # doctest: +SKIP
+        >>> value = metric(reference, hypothesis)       # doctest: +SKIP
+
+    See Also
+    ----------
+    :class:`pyannote.metric.base.BaseMetric`: details on accumulation
+    :class:`pyannote.metric.identification.IdentificationErrorRate`: identification error rate
+    :class:`pyannote.metrics.diarization.DiarizationErrorRate`: diarization error rate
+    """
+
+    @classmethod
+    def metric_name(cls):
+        return WDER
+
+    @classmethod
+    def metric_components(cls):
+        return [WDER, WDER_INCORRECT, WDER_TOTAL,
+                WDER_SPEAKER_SWITCH, WDER_SPEAKER_SWITCH_INCORRECT, WDER_SPEAKER_SWITCH_TOTAL]
+
+    def __init__(self, switch_context=3, **kwargs):
+        super().__init__(**kwargs)
+        self.switch_context = switch_context
```

Comment:
For class members variables, I think we should have docstrings explaining them and/or type annotations

Reply:
added, thanks